src/peer.rs

```diff
@@ -1,10 +1,92 @@
+extern crate serde;
+extern crate serde_json;
 use crate::event::event_hash::EventHash;
 use failure::Error;
+use std::sync::RwLock;
+use std::fs::File;
+use std::io::Read;
 
 pub type PeerId = Vec<u8>;
 
+#[derive(Serialize, Deserialize)]
+pub struct LachesisPeerStruct {
+    #[serde(rename="PubKeyHex")]
+    id: PeerId,
+    #[serde(rename="NetAddr")]
+    net_addr: String,
+    #[serde(skip)]
+    used: i64,
+    #[serde(skip)]
+    height: i64,
+    #[serde(skip)]
+    in_degree: i64,
+    #[serde(skip)]
+    lock: RwLock<()>,
+}
+
+pub trait LachesisPeer {
+    fn new(&self, id: PeerId, net_addr: String) -> LachesisPeerStruct;
+    fn set_height(&mut self, height: i64);
+    fn get_height(&self) -> i64;
+    fn next_height(&mut self) -> i64;
+    fn set_in_degree(&mut self, height: i64);
+    fn get_in_degree(&self) -> i64;
+    fn inc_in_degree(&mut self);
+    fn get_peers(json_peer_path: String) -> Vec<LachesisPeerStruct>;
```

Comment:
Maybe rename `get_local_peers`, for random participation support

Reply:
`get_peers_from_file` ?